pandas/core/arrays/arrow/array.py

```diff
@@ -493,7 +493,21 @@ def _box_pa_array(
             if pa.types.is_dictionary(pa_type):
                 pa_array = pa_array.dictionary_encode()
             else:
-                pa_array = pa_array.cast(pa_type)
+                try:
+                    pa_array = pa_array.cast(pa_type)
+                except (
+                    pa.ArrowInvalid,
+                    pa.ArrowTypeError,
+                    pa.ArrowNotImplementedError,
+                ):
+                    if pa.types.is_string(pa_array.type) or pa.types.is_large_string(
```

Comment:
looking at the pa.types namespace i also see is_unicode, is_large_unicode.  should either of those be handled here?

Reply:
According to the docs, these are aliases of the above so I think it's OK not to include